Check if WebNN op is supported on the platform

zesongw opened the pull request:
- Add a temp builder in helper
- Map the ONNX Optype and WebNN op by the map in helper.h
[issues/1140](https://github.com/otcshare/webnn-native/issues/1140)

Commit:
Add op check for platform

Commit:
Add device and power flags

Honry commented on onnxruntime/core/providers/webnn/builders/helper.cc:
```diff
@@ -68,10 +72,30 @@ std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_v
   std::vector<size_t> supported_node_group;
   const auto& node_indices = graph_viewer.GetNodesInTopologicalOrder();
+
+  // Create WebNN context and graph builder
+  std::unordered_map<uint32_t, std::string> device_type_name_s = {
+    {0, "auto"}, {1, "gpu"}, {2, "cpu"}};
+  std::unordered_map<uint32_t, std::string> power_preference_name_s = {
+      {0, "auto"}, {1, "high-performance"}, {2, "low-power"}};
+  std::string device_type_name_ = device_type_name_s[webnn_device_flags_];
+  std::string power_preference_name_ = power_preference_name_s[webnn_power_flags_];
+  thread_local const emscripten::val ml = emscripten::val::global("navigator")["ml"];
+  emscripten::val context_options = emscripten::val::object();
+  context_options.set("deviceType", emscripten::val(device_type_name_));
+  context_options.set("powerPreference", emscripten::val(power_preference_name_));
+  emscripten::val wnn_context_ = ml.call<emscripten::val>("createContextSync", context_options);
+  emscripten::val wnn_builder_ = emscripten::val::global("MLGraphBuilder").new_(wnn_context_);
+
```
Maybe we can move webnn builder creation to webnn_execution_provider.cc.

Honry commented on onnxruntime/core/providers/webnn/builders/helper.cc:
```diff
@@ -68,10 +72,30 @@ std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_v
     const auto* node(graph_viewer.GetNode(node_idx));
-    bool supported = IsNodeSupported(*node, graph_viewer, logger);
+    bool supported = false;
+    // First check if platform support the op
```
```suggestion
    // Firstly check if platform supports the WebNN op
```

Commit:
Move wnn_context_ and wnn_builder_

Move wnn_context_ and wnn_builder_  to webnn_execution_provider.
Remove the unused flags.

Honry commented on onnxruntime/core/providers/webnn/builders/helper.h:
```diff
@@ -22,8 +25,28 @@ bool GetShape(const NodeArg& node_arg, std::vector<int64_t>& shape, const loggin
 // Get a list of groups of supported nodes, each group represents a subgraph supported by WebNN EP
-std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_viewer,
-                                                      const logging::Logger& logger);
+std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_viewer, const emscripten::val& wnn_builder_, const logging::Logger& logger);
```
Pls. wrap the last two params into lines.

Honry commented on onnxruntime/core/providers/webnn/webnn_execution_provider.cc:
```diff
@@ -35,6 +34,29 @@ WebNNExecutionProvider::WebNNExecutionProvider(uint32_t webnn_device_flags, uint
 
   InsertAllocator(CreateAllocator(cpu_memory_info));
+
+    // Create WebNN context and graph builder
```
Inconsistent indentation.

Honry commented on onnxruntime/core/providers/webnn/webnn_execution_provider.cc:
```diff
@@ -35,6 +34,29 @@ WebNNExecutionProvider::WebNNExecutionProvider(uint32_t webnn_device_flags, uint
+  std::unordered_map<uint32_t, std::string> device_type_name_s = {
+    {0, "auto"}, {1, "gpu"}, {2, "cpu"}};
+  std::unordered_map<uint32_t, std::string> power_preference_name_s = {
+      {0, "auto"}, {1, "high-performance"}, {2, "low-power"}};
```
Ditto.

Honry reviewed (approved):
Generally LGTM, leave a few nits.

Honry commented:
@huningxin, would you like to take another look?

huningxin reviewed (approved):
LGTM, thanks for the PR @zesongw and the review @Honry !

Commit:
Fix nits
